File: aon-occam/src/main/java/com/esferalia/aon/occam/api/model/project/ProjectHolder.java
package com.esferalia.aon.occam.api.model.project;

import java.io.Serializable;
import java.util.Date;

import com.esferalia.aon.occam.api.model.Workgroup;
import com.esferalia.aon.occam.api.model.task.TaskHolder;

public class ProjectHolder implements Serializable {

	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	
	private Integer id;
	private Integer domain;
	private Integer project;
	private Date startDate;
	private Date endDate;
	private Workgroup workgroup;
	private TaskHolder taskHolder;
	
	public ProjectHolder() {
		// Nothing to do.
	}

	public Integer getId() {
		return id;
	}

	public ProjectHolder setId(Integer id) {
		this.id = id;
		return this;
	}

	public Integer getDomain() {
		return domain;
	}

	public ProjectHolder setDomain(Integer domain) {
		this.domain = domain;
		return this;
	}

	public Integer getProject() {
		return project;
	}

	public ProjectHolder setProject(Integer project) {
		this.project = project;
		return this;
	}

	public Date getStartDate() {
		return startDate;
	}

	public ProjectHolder setStartDate(Date startDate) {
		this.startDate = startDate;
		return this;
	}

	public Date getEndDate() {
		return endDate;
	}

	public ProjectHolder setEndDate(Date endDate) {
		this.endDate = endDate;
		return this;
	}

	public Workgroup getWorkgroup() {
		if(workgroup == null) {
			workgroup = new Workgroup();
		}
		return workgroup;
	}

	public ProjectHolder setWorkgroup(Workgroup workgroup) {
		this.workgroup = workgroup;
		return this;
	}

	public TaskHolder getTaskHolder() {
		if(taskHolder == null) {
			taskHolder = new TaskHolder();
		}
		return taskHolder;
	}

	public ProjectHolder setTaskHolder(TaskHolder taskHolder) {
		this.taskHolder = taskHolder;
		return this;
	}
	
	public boolean isEmpty() {
		return getId() == null && getEndDate() == null
				&& getStartDate() == null && getTaskHolder().isEmpty()
				&& getWorkgroup().isEmpty();
				//|| getProject().isEmpty() 
	}
	
}

File: aon-solutions/src/main/webapp/models/project/ProjectHolder.js
import * as LS from '../../services/localStorageService.js';
import { TaskHolder } from './TaskHolder.js';
import { Workgroup } from './Workgroup.js';

export class ProjectHolder {
    id;
    domain;
    project;
    startDate;
    endDate;
    workgroup;
    taskHolder;
    dirty;

    constructor(projectHolder) {
        if(projectHolder) {
            this.id = projectHolder.id;
            this.domain = projectHolder.domain || LS.getDomainId();
            this.project = projectHolder.project;
            this.startDate = projectHolder.startDate || new Date();
            this.endDate = projectHolder.endDate;
            this.workgroup = new Workgroup(projectHolder.workgroup);
            this.taskHolder = new TaskHolder(projectHolder.taskHolder);
            this.dirty = projectHolder.dirty;
        } else {
            this.domain = LS.getDomainId();
            this.startDate = new Date();
            this.workgroup = new Workgroup();
            this.taskHolder = new TaskHolder();
            this.dirty = false;
        }
    }

    getId() {
        return this.id;
    }

    setId(id) {
        this.id = id;
        return this;
    }

    getDomain() {
        return this.domain;  
    }

    setDomain(domain) {
        this.domain = domain;
        return this;
    }

    getProject() {
        return this.project;
    }

    setProject(project) {
        this.project = project;
        return this;
    }

    getStartDate(){
        return this.startDate;
    }
 
    setStartDate(startDate) {
        this.startDate = startDate;
        return this;
    }

    getEndDate() {
        return this.this.endDate;
    }

    setEndDate(endDate) {
        this.endDate = endDate()
        return this;
    }

    getWorkgroup() {
        return this.workgroup;
    }

    setWorkgroup(workgroup){
        this.workgroup = new Workgroup(workgroup);
        return this;
    }

    getTaskHolder() {
        return this.taskHolder;
    }

    setTaskHolder(taskHolder) {
        this.taskHolder = new TaskHolder(taskHolder);
        return this;
    }

    isDirty() {
        return this.dirty;
    }

    setDirty(dirty) {
        this.dirty = dirty;
        return this;
    }
} 